src/s1reader/s1_orbit.py

```diff
@@ -1,12 +1,169 @@
+import cgi
 import datetime
 import glob
 import os
+import requests
 import warnings
 
-
 # date format used in file names
 FMT = "%Y%m%dT%H%M%S"
 
+# Required for orbit download
+scihub_url = 'https://scihub.copernicus.eu/gnss/odata/v1/Products'
+# Namespaces of the XML file returned by the S1 query. Will they change it?
+w3_url = '{http://www.w3.org/2005/Atom}'
+m_url = '{http://schemas.microsoft.com/ado/2007/08/dataservices/metadata}'
+d_url = '{http://schemas.microsoft.com/ado/2007/08/dataservices}'
+# Scihub guest credential
+scihub_user = 'gnssguest'
+scihub_password = 'gnssguest'
+
+
+def check_internet_connection():
+    '''
+    Check connection availability
+    '''
+    url = "http://google.com"
+    try:
+        requests.get(url, timeout=10)
+    except (requests.ConnectionError, requests.Timeout) as exception:
+        raise sys.exit(exception)
+
+
+def parse_safe_filename(safe_filename):
```

Comment:
There's enough overlap/redundancy between `parse_safe_filename` and [`get_file_name_tokens`](https://github.com/opera-adt/s1-reader/blob/970ea7ed2d4aa56fe642d63010a7b7db09efb63c/src/s1reader/s1_orbit.py#L167) that should be combined. May be better to put this in a catch-all src/s1reader/utils.py for easy importing/sharing elsewhere?

Reply:
Not sure whether we will need this functionality to be imported elsewhere. I am fine with the combination of the two functions. Can the combined function have as returned value the one returned by `parse_safe_filename`?

Reply:
I think resolving the overlapped functionality should be in different PR. I will open an issue. IIRC the noise and calibration reader has implemented it's own as well. Reminds me of [this scene](https://youtu.be/Et4sMJP9FmM?t=128) from the movie Contact.